reana_workflow_engine_snakemake/executor.py

```diff
@@ -154,55 +116,112 @@ def _get_container_image(self, job: Job) -> str:
             log.info(f"No environment specified, falling back to: {container_image}")
         return container_image
 
-    def _handle_job_status(self, job, status):
+    def _handle_job_status(
+        self, job: Job, job_status: JobStatus, workflow_status: RunStatus
+    ) -> None:
         workflow_uuid = os.getenv("workflow_uuid", "default")
         job_id = job.reana_job_id
-        log.info(f"{status} job: {job_id}")
+        log.info(f"{job_status} job: {job_id}")
         message = {
             "progress": build_progress_message(
-                **{status: {"total": 1, "job_ids": [job_id]}}
+                **{job_status.name: {"total": 1, "job_ids": [job_id]}}
             )
         }
-        status_running = 1
-        status_failed = 3
-        status_mapping = {"finished": status_running, "failed": status_failed}
         self.publisher.publish_workflow_status(
-            workflow_uuid, status_mapping[status], message=message
+            workflow_uuid, workflow_status.value, message=message
         )
 
-    def handle_job_success(self, job):
+    def handle_job_success(self, job: Job) -> None:
         """Override job success method to publish job status."""
         # override handle_touch = True, to enable `touch()` in Snakefiles
+        # `touch()` is responsible for checking output files existence
```

Comment:
Not really, `touch()` is just the Unix `touch` that creates empty files, so it can denote that a certain task was completed.

Reply:
If I will disable the `touch()` (`handle_touch=True`) then the CMS-h4l demo is failing.

Reply:
Found `handle_touch` function in Snakemake - https://github.com/snakemake/snakemake/blob/987282dde8a2db5174414988c134a39ae8836a61/snakemake/dag.py#L580

Reply:
I meant the docstring, it doesn't seem 100% accurate to me. No comments about functionality, we need to keep `handle_touch=True` as you've described.

```
# `touch()` is responsible for checking output files existence
```

Reply:
So, what do you propose, remove the comment or rewrite it to be more accurate?